Remove support for Ubuntu 17.10

rallytime opened the pull request:
Ubuntu 17.10 will be EOL any day now, as Ubuntu's release calendar says support will end in July 2018.

This commit removes support for the 17.10 (non-LTS) release. The README file was also updated to clarify when support for non-LTS releases will be removed from bootstrap.

**DO NOT MERGE UNTIL 17.10 IS OFFICIALLY EOL**

Release schedule: https://wiki.ubuntu.com/Releases

Commit:
Remove support for Ubutu 17.10

Ubuntu 17.10 will be EOL any day now, as Ubuntu's release calendar
says support will end in July 2018.

This commit removes support for the 17.10 (non-LTS) release. The
README file was also updated to clarify when support for non-LTS
releases will be removed from bootstrap.

KaiSforza reviewed (approved):
:+1:

vutny commented on bootstrap-salt.sh:
```diff
@@ -2576,11 +2573,7 @@ __enable_universe_repository() {
-        UBUNTU_VERSION=16.04
-        UBUNTU_CODENAME="xenial"
-    elif [ "$DISTRO_MAJOR_VERSION" -gt 18 ]; then
+    if [ "$DISTRO_MAJOR_VERSION" -gt 18 ]; then
```
Probably we would like the Bootstrap to be more future-proof and better detect some-day upcoming Ubuntu release 18.10. Something like this will be more robust:
```shell
    if [ "$DISTRO_MAJOR_VERSION" -gt 18 ] || [ "${DISTRO_VERSION}" = "18.10" ]; then
```

vutny commented on README.rst:
```diff
@@ -289,6 +289,9 @@ release instead.
 For example, when installing Salt on Ubuntu 17.10, the bootstrap script will setup the repository
```
While deprecating support for 17.10, maybe let's remove mentioning it and switch over number 18, what do you think?

vutny reviewed (changes requested):
This is great @rallytime .
I just have couple of minor things to suggest.

Commit:
Make some initial preparations for Ubuntu 18.10 support

Ubuntu 18.10 is not released yet, but we can make some small changes now
to make the script a little more robust for when it is released.

rallytime commented:
@vutny Yes, good suggestions. I've incorporated them in the latest commit.

Commit:
Merge branch 'develop' into remove-ubuntu-17
